src/main/java/io/debezium/connector/cassandra/CommitLogReadHandlerImpl.java

```diff
@@ -303,18 +303,22 @@ private void process(PartitionUpdate pu, OffsetPosition offsetPosition, Keyspace
      * Handle a valid deletion event resulted from a partition-level deletion by converting Cassandra representation
      * of this event into a {@link Record} object and queue the record to {@link ChangeEventQueue}. A valid deletion
      * event means a partition only has a single row, this implies there are no clustering keys.
-     *
+     * <p>
      * The steps are:
-     *      (1) Populate the "source" field for this event
-     *      (2) Fetch the cached key/value schemas from {@link SchemaHolder}
-     *      (3) Populate the "after" field for this event
-     *          a. populate partition columns
-     *          b. populate regular columns with null values
-     *      (4) Assemble a {@link Record} object from the populated data and queue the record
+     * (1) Populate the "source" field for this event
+     * (2) Fetch the cached key/value schemas from {@link SchemaHolder}
+     * (3) Populate the "after" field for this event
+     * a. populate partition columns
+     * b. populate regular columns with null values
+     * (4) Assemble a {@link Record} object from the populated data and queue the record
      */
     private void handlePartitionDeletion(PartitionUpdate pu, OffsetPosition offsetPosition, KeyspaceTable keyspaceTable) {
 
-        SchemaHolder.KeyValueSchema keyValueSchema = schemaHolder.getOrUpdateKeyValueSchema(keyspaceTable);
+        SchemaHolder.KeyValueSchema keyValueSchema = schemaHolder.getKeyValueSchema(keyspaceTable);
+        if (keyValueSchema == null) {
```

Comment:
Imagine there is a bunch of commit logs in cdc and your processor / handler are reading them (Mutations) but at the same time somebody dropped that table via cql. So what happens is that a listener is invoked and it would be removed from the holder but you might still read a Mutation which was just on cdc before but it does not exist anymore and holder does not hold it either, but you are still trying to send it to Kafka.

Reply:
Nice catch! Can you also add a log here please? Something like:
```
LOGGER.warn("Unable to get KeyValueSchema for table {}. It might have been deleted or CDC disabled.", keyspaceTable.toString()).
```